packages/flutter_tools/lib/src/commands/emulators.dart

```diff
@@ -16,6 +16,8 @@ class EmulatorsCommand extends FlutterCommand {
   EmulatorsCommand() {
     argParser.addOption('launch',
         help: 'The full or partial ID of the emulator to launch.');
+    argParser.addOption('create',
```

Comment:
Should `--create` be a flag (instead of an option)? We could auto-name the resulting image (`pixel`, `pixel_xl-2`, ...), and have an optional flag to name a created image.

Reply:
`create` feels like a subcommand, even if the `package:args` api doesn't make it easy to have it as one.

Reply:
> Should `--create be a flag` (instead of an option)?  We could auto-name the resulting image

I was a bit unsure about this. If we pick the name for the user then it feels weird if we tell them "an emulator with that name already exists" so I was going to make them always provide it.

Of course, we could do the check ourselves first and only prompt if it already exists, but that might be better handled in the editor than here so it has better control over how to present it.

I was wondering if we should expose an `emulator.getDeviceProfiles`. That way when the user chooses "create emulator" in the editor, we can fetch the possible profiles (and filter it to pixels, nexuses, whatever) and then let them pick a device. I think being able to pick between a phone and a tablet could be helpful?

Reply:
> If we pick the name for the user then it feels weird if we tell them "an emulator with that name already exists"

We could choose a name, see if it already exists, and suffix something to de-dup (my_name-2).

> and then let them pick a device

I generally like to avoid asking the user for input on stdin. I think `flutter emulators --create --name=my_device` would be a reasonable UI, but am not fixed on it if you think there's a better way.

Reply:
Ok, gone with this. We only add suffixes if we're generating the name, so if the user provides an explicit name we'll just give them an error:

```
dantup-macbookpro:flutter dantup$ flutter emulator --create
Emulator 'flutter_emulator' created successfully.

dantup-macbookpro:flutter dantup$ flutter emulator --create
Emulator 'flutter_emulator_1' created successfully.

dantup-macbookpro:flutter dantup$ flutter emulator --create --name flutter_emulator_1
Failed to create emulator 'flutter_emulator_1'.
Error: Android Virtual Device 'flutter_emulator_1' already exists.
```

It feels a bit weird doing `flutter emulator --create` and not `flutter emulator create` but I think that's unavoidable as-is.

Do we have any convention for indicating when a flag is only usable in conjunction with another (eg. `--name` only makes sense with `--create`)? I couldn't see anything obvious.

Reply:
> Do we have any convention for indicating when a flag is only usable in conjunction with another (eg. --name only makes sense with --create)? I couldn't see anything obvious

Not that I know of; likely just something we should do w/ the help text for the flags.

Reply:
I found some are prefixed with `Used with flag --xxx.` so I'll copy that.